network/src/protocols/network/mod.rs

```diff
@@ -172,22 +174,58 @@ pub trait NewNetworkEvents {
     fn new(
         peer_mgr_notifs_rx: aptos_channel::Receiver<(PeerId, ProtocolId), PeerManagerNotification>,
         connection_notifs_rx: aptos_channel::Receiver<PeerId, ConnectionNotification>,
+        bounded_executor: BoundedExecutor,
     ) -> Self;
 }
 
-impl<TMessage: Message> NewNetworkEvents for NetworkEvents<TMessage> {
+impl<TMessage: Message + Send + 'static> NewNetworkEvents for NetworkEvents<TMessage> {
```

Comment:
Why does the TMessage need to be static?

Reply:
This is required to satisfy the `move` call in `spawn_blocking`, e.g., see: https://tokio.rs/tokio/tutorial/spawning